### ISSUE TITLE
Alias for default tag name broken for @constructor

### ISSUE THREAD

**Issue by Nyphet** [gajus/eslint-plugin-jsdoc#785]
## Expected behavior
Not to give off a warning saying "Invalid JSDoc tag (preference). Replace "constructor" JSDoc tag with "class"." when I set 
`"settings": { "jsdoc": { "tagNamePreference": { "constructor": "constructor" } }`
For other tag aliases, like the one proposed in the README "return", it works fine.

## Actual behavior
![image](https://user-images.githubusercontent.com/65339023/134389238-045c6f5b-51e2-4e59-a09a-d93bf95a5944.png)

## ESLint Config

```json
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "jsdoc"
  ],
  "rules": {
    "jsdoc/check-tag-names": "warn"
  },
  "settings": {
    "jsdoc": {
      "tagNamePreference": {
        "constructor": "constructor"
      }
    }
  },
  "ignorePatterns": [
    "build",
    "dist",
    "**/*.d.ts"
  ]
}
```

## ESLint sample

```ts
/**
 * @constructor
 */
 constructor() {}
```

## Environment

- Node version: 14.17.1
- ESLint version: 7.32.0
- `eslint-plugin-jsdoc` version: 36.1.0

**Comment by brettz9** [comment]
Are you only getting this in your IDE or also when you run the `eslint` binary against it?

I'm not able to replicate with the binary using those settings and the following code:

```js
      class A {
        /**
         * @constructor
         */
         constructor() {
           super();
         }
      }
```

**Comment by brettz9** [comment]
Yes, if you are getting the problem from the command line like that, then there does appear to be some problem. (unless it is due to an older ESLint version which I wouldn't think).

Could you make a repository which minimally reproduces the problem? I.e., which has nothing but a single simple JavaScript/TypeScript file, package.json, and ESLint config file that can reproduce the problem?

**Comment by Nyphet** [comment]
Basic repository to reproduce the error created [here](https://github.com/Nyphet/eslint-test-prj).

**Comment by brettz9** [comment]
Ok, apologies for the long delay... As per [the docs](https://github.com/gajus/eslint-plugin-jsdoc#alias-preference), it turns out, you just need to use `tag constructor`:

```js
      settings: {
        jsdoc: {
          tagNamePreference: {
            'tag constructor': 'constructor',
          },
        },
      },
```

Closing as that should resolve, though feel free to comment further.

**Comment by tschaub** [comment]
A related issue is that you can no longer use the `@constructor` tag at all if you happen to have any `tagNamePreference` set (even if you don't use an alias for `constructor`).  See #898.
